businessRules/users.js

```diff
@@ -0,0 +1,56 @@
+const { v4 } = require('uuid');
+const md5 = require('md5');
+const { createUser } = require('../models/users');
+const { getUsers, setUsers } = require('../utils/storage');
+
+function registerUser(userName, password, firstName, lastName) {
+    const noFirstName = 'Anonymous';
+    const noLastName = 'Incognito';
+
+    if (!firstName) {
+        firstName = noFirstName;
+    }
+
+    if (!lastName) {
+        lastName = noLastName;
+    }
+
+    createUser(userName, password, firstName, lastName);
+}
+
+function login(req, res) {
+    let users = getUsers();
+
+    const userName = req.body.userName;
+    const password = md5(req.body.password);
+
+    const user = users.find(
+        (u) => u.userName === userName && u.password === password
+    );
+
+    if (user) {
+        const sessionId = md5(v4()); //Should be REAL cryptography.
+        user.session = sessionId;
+        setUsers(users);
+        res.cookie('userLoginSession', sessionId, {
+            sameSite: 'None',
+            secure: true,
+            httpOnly: true,
+            maxAge: 900000,
+        });
+        res.status(200).json({ status: 'OK', username: user.userName });
+    } else {
+        res.status(400).json({ status: 'No such user.' });
+    }
```

Comment:
Handle error state first.

Check if user doesn't exist and handle that case first.

Then handle the success case.

```js
if (!user) {
  res.status(400).json({ status: 'No such user.' });
  return; // This return statement removes the need for `else` statement after this `if`
}
```

Reply:
Updated and tested.